src/qibo/tests/test_backends_agreement.py

```diff
@@ -50,7 +50,7 @@ def test_backend_methods_list(tested_backend, target_backend, method, args):
     tested_func = getattr(tested_backend, method)
     target_func = getattr(target_backend, method)
     target_result = target_func(*args)
-    if tested_backend.name == "qibojit" and tested_backend.engine.name in ["cupy", "cuquantum"]: # pragma: no cover
+    if tested_backend.name == "qibojit" and tested_backend.platform.name in ["cupy", "cuquantum"]: # pragma: no cover
```

Comment:
This change is done in #533, in fact I completely removed this if and updated the casting below so that it works with all backends/platforms and we don't have to modify it every time we update something in qibojit.